tests/src/test/scala/com/nvidia/spark/rapids/AdaptiveQueryExecSuite.scala

```diff
@@ -90,6 +99,89 @@ class AdaptiveQueryExecSuite
     }
   }
 
+  test("Plugin should translate child plan of GPU DataWritingCommandExec to GPU") {
+
+    val conf = new SparkConf()
+        .set(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key, "true")
+        .set(SQLConf.ADAPTIVE_EXECUTION_FORCE_APPLY.key, "true")
+
+    withGpuSparkSession(spark => {
+      import spark.implicits._
+
+      // read from a parquet file so we can test reading on GPU
+      val path = new File(TEST_FILES_ROOT, "DataWritingCommandExecGPU.parquet").getAbsolutePath
+      (0 until 100).toDF("a")
+          .write
+          .mode(SaveMode.Overwrite)
+          .parquet(path)
+      spark.read.parquet(path).createOrReplaceTempView("testData")
+
+      spark.sql("CREATE TABLE IF NOT EXISTS DataWritingCommandExecGPU (a INT) USING parquet")
+          .collect()
+
+      val df = spark.sql("INSERT INTO TABLE DataWritingCommandExecGPU SELECT * FROM testData")
+      df.collect()
+
+      // write should be on GPU
+      val writeCommand = TestUtils.findOperator(df.queryExecution.executedPlan,
+        _.isInstanceOf[GpuDataWritingCommandExec])
+      assert(writeCommand.isDefined)
+
+      // the read should be an adaptive plan
+      val adaptiveSparkPlanExec = TestUtils.findOperator(writeCommand.get,
+        _.isInstanceOf[AdaptiveSparkPlanExec])
+        .get.asInstanceOf[AdaptiveSparkPlanExec]
+
+      // assert that at least part of the adaptive plan ran on GPU
+      assert(TestUtils.findOperator(adaptiveSparkPlanExec, _.isInstanceOf[GpuExec]).isDefined)
+    }, conf)
+  }
+
+  test("Plugin should translate child plan of CPU DataWritingCommandExec to GPU") {
+
+    val conf = new SparkConf()
+      .set(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key, "true")
+      .set(SQLConf.ADAPTIVE_EXECUTION_FORCE_APPLY.key, "true")
+      // force DataWritingCommandExec onto CPU for this test because we want to verify that
+      // the read will still happen on GPU with a CPU write
+      .set(RapidsConf.TEST_ALLOWED_NONGPU.key, "DataWritingCommandExec")
+      .set("spark.rapids.sql.exec.DataWritingCommandExec", "false")
+
+      withGpuSparkSession(spark => {
+        import spark.implicits._
+
+        // read from a parquet file so we can test reading on GPU
+        val path = new File(TEST_FILES_ROOT, "DataWritingCommandExecCPU.parquet").getAbsolutePath
+        (0 until 100).toDF("a")
+            .write
+            .mode(SaveMode.Overwrite)
+            .parquet(path)
```

Comment:
I think the test should clean up after itself, maybe an after method that cleans up TEST_FILES_ROOT?